### PR TITLE
chore: refactoring installed app read service

### PR DESCRIPTION
**Pull request by Ash-exp**
# Description

Fixes #

## Checklist:

* [ ] The title of the PR states what changed and the related issues number (used for the release note).
* [ ] Does this PR requires documentation updates?
* [ ] I've updated documentation as required by this PR.
* [ ] I have performed a self-review of my own code.
* [ ] I have commented my code, particularly in hard-to-understand areas.
* [ ] I have tested it for all user roles.
* [ ] I have added all the required unit/api test cases.

## Does this PR introduce a user-facing change?

```release-note

```

### CODE REVIEW

File: Wire.go [hunk 1]
```diff
@@ -538,12 +539,17 @@ func InitializeApp() (*App, error) {
...
 		appStoreRestHandler.NewAppStoreStatusTimelineRestHandlerImpl,
 		wire.Bind(new(appStoreRestHandler.AppStoreStatusTimelineRestHandler), new(*appStoreRestHandler.AppStoreStatusTimelineRestHandlerImpl)),
+
 		appStoreRestHandler.NewInstalledAppRestHandlerImpl,
 		wire.Bind(new(appStoreRestHandler.InstalledAppRestHandler), new(*appStoreRestHandler.InstalledAppRestHandlerImpl)),
+
 		FullMode.NewInstalledAppDBExtendedServiceImpl,
 		wire.Bind(new(FullMode.InstalledAppDBExtendedService), new(*FullMode.InstalledAppDBExtendedServiceImpl)),
+
+		installedAppReader.WireSet,
```
Comment (kartik-579) [inline Wire.go:552]: move this into the common wire set

File: pkg/appStore/installedApp/read/adapter/adapter.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+	if !installedAppModel.Environment.IsZero() {
+		installedAppInternal.EnvironmentName = installedAppModel.Environment.Name
+		installedAppInternal.EnvironmentIdentifier = installedAppModel.Environment.EnvironmentIdentifier
+		installedAppInternal.Namespace = installedAppModel.Environment.Namespace
+		installedAppInternal.ClusterId = installedAppModel.Environment.ClusterId
+	}
+	// Cluster details
+	if !installedAppModel.Environment.Cluster.IsZero() {
+		installedAppInternal.ClusterName = installedAppModel.Environment.Cluster.ClusterName
+	}
```
Comment (kartik-579) [inline pkg/appStore/installedApp/read/adapter/adapter.go:34]: please double check if any other fields were used earlier

File: api/appStore/deployment/wire_appStoreDeployment.go [hunk 3]
```diff
@@ -27,7 +28,7 @@ import (
 	"github.com/google/wire"
 )
 
-var AppStoreDeploymentWireSet = wire.NewSet(
+var WireSet = wire.NewSet(
```
Comment (kartik-579) [inline api/appStore/deployment/wire_appStoreDeployment.go:31]: have two wire sets here for EA and full mode and let's keep installedAppReader.* in these sets

File: pkg/appStore/installedApp/read/bean/bean.go [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+	IsCustomRepository bool
+	// Deprecated; use deployment_config table instead
+	DeploymentAppType          string
+	DeploymentAppDeleteRequest bool
+	EnvironmentId              int
+	AppId                      int
+}
+
+type InstalledAppWithAppDetails struct {
+	InstalledAppMin
```
Comment (kartik-579) [inline pkg/appStore/installedApp/read/bean/bean.go:20]: try to keep all nested objects as pointers

**Review by kartik-579 — CHANGES_REQUESTED** [review]

File: pkg/appStore/installedApp/read/adapter/adapter.go (hunk 2 above)
Comment (Ash-exp) [reply pkg/appStore/installedApp/read/adapter/adapter.go:34]: as I have replaced the response type from `installedAppModel` to `installedAppInternal`. So I've also updated the usage of installedAppModel and added the required fields to installedAppInternal struct.

File: api/appStore/deployment/wire_appStoreDeployment.go (hunk 3 above)
Comment (Ash-exp) [reply api/appStore/deployment/wire_appStoreDeployment.go:31]: installedAppReader also has two wire sets. so they should be registered accordingly. AppStoreDeploymentWireSet is common for both EA and FULL mode. so registering `installedAppReader.EAWireSet` here

File: Wire.go (hunk 1 above)
Comment (Ash-exp) [reply Wire.go]: updated 👍

File: pkg/appStore/installedApp/read/bean/bean.go (hunk 4 above)
Comment (Ash-exp) [reply pkg/appStore/installedApp/read/bean/bean.go]: updated 👍

File: pkg/appStore/installedApp/read/InstalledAppReadService.go [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+	"github.com/devtron-labs/devtron/pkg/appStore/installedApp/read/adapter"
+	"github.com/devtron-labs/devtron/pkg/appStore/installedApp/read/bean"
+)
+
+type InstalledAppReadService interface {
+	InstalledAppReadServiceEA
+	// GetInstalledAppByGitHash will return the installed app by git hash.
+	// Only delete specific details are fetched.
+	// Refer bean.InstalledAppDeleteRequest for more details.
+	GetInstalledAppByGitHash(gitHash string) (*bean.InstalledAppDeleteRequest, error)
```
Comment (nishant-d) [inline pkg/appStore/installedApp/read/InstalledAppReadService.go:13]: repo_url

File: pkg/workflow/status/WorkflowStatusService.go [hunk 6]
```diff
@@ -334,12 +338,8 @@ func (impl *WorkflowStatusServiceImpl) UpdatePipelineTimelineAndStatusByLiveAppl
 			impl.logger.Errorw("error in getting envDetails from environment id", "err", err)
 			return nil, isTimelineUpdated
 		}
-		var acdAppName string
-		if len(installedApp.Environment.Name) != 0 {
-			acdAppName = appDetails.AppName + installedApp.Environment.Name
-		} else {
-			acdAppName = appDetails.AppName + "-" + envDetail.Name
-		}
+		// ArgoCD application nam: format: <appName>-<envName>
```
Comment (nishant-d) [inline pkg/workflow/status/WorkflowStatusService.go:341]: use. util

**Review by nishant-d — CHANGES_REQUESTED** [review]

**Review by nishant-d — APPROVED** [review]

**Review by kartik-579 — APPROVED** [review]

File: pkg/appStore/installedApp/read/InstalledAppReadService.go (hunk 5 above)
Comment (Ash-exp) [reply pkg/appStore/installedApp/read/InstalledAppReadService.go:13]: NOTE: 
`GetInstalledAppByGitHash(gitHash string)` : handling just with gitHash is not valid for different repositories. although the possibility is very low, but not impossible. 
This'll be a minor change from code side to update the query to verify the repository url as well. but as this is a logical change will be handling in a different ticket.

cc: @vikramdevtron

**Pull request merged**